divide_and_conquer/quickhull.py

```diff
@@ -0,0 +1,89 @@
+'''
+Quick-hull-
+Created on 22/7/2020
+---description---
+Quick-hull is a divide and conquer algorithm designed to construct a convex hull.
+it is called Quick-hull as its logic is closer to Quick-sort.
+Complexity: O(n**2)
+---------input---------
+-->list of points in list format
+---------output--------
+--> list of end-points
+>>> quickhull(a = [[1,3],[1,1],[2,2],[4,4],[0,0],[1,2],[3,1],[3,3]])
+the hull endpoints are  [[0, 0], [1, 3], [4, 4], [3, 1]]
+
+>>> quickhull(a = [[1,3],[1,1],[2,2],[4,4],[0,0],[1,2],[3,1],[3,3],[1,5],[3,2],[5,5],[4,0]])
+the hull endpoints are  [[0, 0], [1, 5], [5, 5], [4, 0]]
+
+>>> quickhull(a = ['lol'])
+check the input format: its should be list of lists: [[5,6],[4,8]...]
+given format:  ['lol']
+'''
+
+def min_max_x(a):

```

Comment:
```suggestion
from typing import List, Tuple


def min_max_x(points: List[List[int, int]]) -> Tuple[int, int]:
```
Add type hints and avoid single letter names for function parameters.